packages/at_auth/lib/src/keys/at_keys_io.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'at_keys.dart' show AtKeys;
+import 'package:at_auth/src/auth_constants.dart' as auth_constants;
+import 'package:at_chops/at_chops.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:at_utils/at_utils.dart' show AtSignLogger;
+
+abstract class AtKeysIo {
+  FutureOr<AtKeys> read(String atSign);
+  FutureOr write(String atSign, AtKeys atKeys);
+}
+
+class BaseAtKeysIo extends AtKeysIo {
```

Comment:
You could probably make this a mixin https://dart.dev/language/mixins

Reply:
e.g. `mixin BaseAtKeysIo on AtKeysIo`.

Then you can drop the unimplemented overrides.